### ISSUE TITLE
GraphQL Support

### ISSUE THREAD

**Issue by SeijiV13** [Mackignacio/maya#15]
## 📝 Description
Suggestion for an enhancement.
-supports graph Ql API
## 📋 Todo:
- [ ] Include Graph QL Support for MayaJS

**Comment by mackignacio** [comment]
What will be a good approach to this? Most likely this is another module that you will import inside a controller. Another approach is a combination of module and decorator. @SeijiV13 if you have an idea kindly make a branch and we can work on this.

**Comment by SeijiV13** [comment]
@Mackignacio I think the best approach would be the combination of module and decorator that will identify what endpoints that will be compatible for GraphQL. I can research on this to have more idea on the best implementation

**Comment by mackignacio** [comment]
@SeijiV13 Ok That's great. I'm also thinking this must be on another repo separate from the core library.

**Comment by SeijiV13** [comment]
@Mackignacio yes great idea, as GraphQL may have additional features like mutations, this is good if it is separated.

**Comment by mackignacio** [comment]
But for now we can do this on a separate branch.
